bundles/action/org.openhab.action.satel/src/main/java/org/openhab/action/satel/internal/Satel.java

```diff
@@ -0,0 +1,190 @@
+
+    /**
+     * Checks current connection status to the communication module.
+     * 
+     * @return <code>true</code> if connection is established
+     */
+    @ActionDoc(text = "Check current connection status to Satel communication module", returns = "<code>true</code>, if connected and <code>false</code> otherwise.")
+    public static boolean satelIsConnected() {
+        if (!SatelActionService.isProperlyConfigured) {
+            logger.debug("Satel action is not yet configured - execution aborted!");
+            return false;
+        }
+        if (SatelActionService.satelCommModule == null) {
+            logger.debug("Satel communication module not available - execution aborted!");
+            return false;
+        }
+        if (!SatelActionService.satelCommModule.isConnected()) {
+            logger.debug("Satel communication module is not connected");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Read event log record from the alarm system for given index.
+     * For the most recent record pass <code>-1</code> as index,
+     * next records can be retrieved by using next index value in the record data (the last field in a record).
+     * 
+     * Fields in the record:
+     * <ol>
+     * <li>timestamp - {@link DateTimeType}</li>
+     * <li>partition - String</li>
+     * <li>event class - {@link EventClass}</li>
+     * <li>event code - Integer}</li>
+     * <li>state restoration flag - Boolean</li>
+     * <li>event description - String</li>
+     * <li>kind of description - Integer</li>
+     * <li>source - Integer</li>
+     * <li>object - Integer</li>
+     * <li>user control number - Integer</li>
+     * <li>next record index - Integer</li>
+     * <li>current record index - Integer</li>
+     * </ol>
+     * 
+     * <p>
+     * The details about each field meaning can be found in protocol documentation.
+     * </p>
+     * 
+     * @param eventIndex event log record index to read
+     * @return event as array of objects
+     */
+    @ActionDoc(text = "Returns record from the event log for given index", returns = "Array of values: timestamp, partition, event class, event code, restore, event description, kind og description, source, object, user control object, next event index, current event index")
+    public static Object[] satelReadEvent(
+            @ParamDoc(name = "eventIndex", text = "index of event to read, -1 for the most recent") int eventIndex) {
+        if (!satelIsConnected()) {
+            return null;
+        }
+        ReadEventCommand cmd = new ReadEventCommand(eventIndex);
+        if (!SatelActionService.satelCommModule.sendCommand(cmd)) {
+            logger.error("Unable to read record for given index: {}", eventIndex);
+            return null;
+        } else if (cmd.isEmpty()) {
+            logger.warn("No record under given index: {}", eventIndex);
```

Comment:
When senCommand returns "false" it does mean a failure, that's why we log an error. When it returns "true", but there is no data, it means that wrong index has been given and warn should be enough to tell the user something is broken in the rule.